Realm/ObjectServerTests/SwiftCollectionSyncTests.swift

```diff
@@ -338,3 +338,602 @@ class CollectionSyncTestCase: SwiftSyncTestCase {
+        let subDict4: AnyRealmValue = AnyRealmValue.fromDictionary([ "key3": subArray4 ])
+        let subArray5: AnyRealmValue = AnyRealmValue.fromArray([ subDict4 ])
+        let subDict5: AnyRealmValue = AnyRealmValue.fromDictionary([ "key4": subArray5 ])
+        let subArray6: AnyRealmValue = AnyRealmValue.fromArray([ subDict5 ])
+        let subDict6: AnyRealmValue = AnyRealmValue.fromDictionary([ "key5": subArray6 ])
+        let dictionary: Dictionary<String, AnyRealmValue> = [
+            "key0": subDict6,
+        ]
+
+        try await write { realm in
+            let object = SwiftTypesSyncObject()
+            object.anyCol = AnyRealmValue.fromDictionary(dictionary)
+            object.stringCol = self.name
+            realm.add(object)
+        }
+
+        let realm = try await openRealm()
+        let results0 =  try await realm.objects(SwiftTypesSyncObject.self).where { $0.stringCol == name }.subscribe()
+        XCTAssertEqual(results0.count, 1)
+
+        let o = results0.first!
+        let baseNested: List<AnyRealmValue>? = o.anyCol.dictionaryValue?["key0"]?.dictionaryValue?["key5"]?.listValue?[0].dictionaryValue?["key4"]?.listValue?[0].dictionaryValue?["key3"]?.listValue
+        let nested1: String? = baseNested?[0].dictionaryValue?["key2"]?.listValue?[0].listValue?[0].object(SwiftPerson.self)?.firstName
+        XCTAssertEqual(nested1, so.firstName)
+        let nested2: AnyRealmValue? = baseNested?[0].dictionaryValue?["key2"]?.listValue?[0].listValue?[1]
+        XCTAssertEqual(nested2, .decimal128(Decimal128(number: 457)))
+        let nested3: String? = baseNested?[0].dictionaryValue?["key2"]?.listValue?[1].dictionaryValue?["key1"]?.listValue?[0].object(SwiftPerson.self)?.firstName
+        XCTAssertEqual(nested3, so.firstName)
+        let nested4: AnyRealmValue? = baseNested?[0].dictionaryValue?["key2"]?.listValue?[1].dictionaryValue?["key1"]?.listValue?[1]
+        XCTAssertEqual(nested4, .decimal128(Decimal128(number: 457)))
+
+        XCTAssertEqual(nested2?.decimal128Value, Decimal128(number: 457))
+        XCTAssertEqual(nested4?.decimal128Value, Decimal128(number: 457))
+    }
+
+    @MainActor func testUpdateMixedList() async throws {
+        let realm1 = try await openRealm()
+        let results1 =  try await realm1.objects(SwiftTypesSyncObject.self).where { $0.stringCol == name }.subscribe()
+
+        let realm2 = try await openRealm()
+        let results2 =  try await realm2.objects(SwiftTypesSyncObject.self).where { $0.stringCol == name }.subscribe()
+
+        // Add initial list
+        try realm1.write {
+            let object = SwiftTypesSyncObject()
+            object.anyCol = AnyRealmValue.fromArray([])
+            object.stringCol = self.name
+            realm1.add(object)
+        }
+        try await realm1.syncSession?.wait(for: .upload)
+        try await realm2.syncSession?.wait(for: .download)
+        try await realm2.syncSession?.wait(for: .upload)
```

Comment:
Waiting for upload completion should not be doing anything here. These tests do need to be calling `await realm2.asyncRefresh()` after waiting for `.download`, though.